third_party/webhook/authz.go

```diff
@@ -87,6 +90,41 @@ func (a *authorizer) clientX509(ctx context.Context) (*client, error) {
 	return newClient(a.ZMSEndpoint, a.ZTSEndpoint, a.Timeout, xpX509), nil
 }
 
+func convertIntoV1(rV1Beta1 authzv1beta1.SubjectAccessReview) authz.SubjectAccessReview {
+	v1Extra := make(map[string]authz.ExtraValue)
+	for key, value := range rV1Beta1.Spec.Extra {
+		v1Extra[key] = authz.ExtraValue(value)
+	}
+
+	return authz.SubjectAccessReview{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       rV1Beta1.Kind,
+			APIVersion: authzSupportedVersion,
+		},
+		ObjectMeta: rV1Beta1.ObjectMeta,
+		Spec: authz.SubjectAccessReviewSpec{
+			User:                  rV1Beta1.Spec.User,
+			UID:                   rV1Beta1.Spec.UID,
+			Extra:                 v1Extra,
+			Groups:                rV1Beta1.Spec.Groups,
+			NonResourceAttributes: (*authz.NonResourceAttributes)(rV1Beta1.Spec.DeepCopy().NonResourceAttributes),
```

Comment:
```suggestion
			NonResourceAttributes: &authz.NonResourceAttributes{
				Path: rV1Beta1.Spec.NonResourceAttributes.Path,
				Verb:      rV1Beta1.Spec.NonResourceAttributes.Verb,
				},
```
DeepCopy is cool too, but new struct looks more consistent. The decision is up to you. 👍

Reply:
handled!